core/templates/services/debouncer.service.ts

```diff
@@ -27,29 +27,25 @@ export class DebouncerService {
   // be invoked. The function only gets executed after it stops being called
   // for `wait` milliseconds.
   debounce(func: () => void, millisecsToWait: number): () => void {
-    let timeout;
-    let context = this;
-    let args = arguments;
-    let timestamp;
-    let result;
+    let timeout: ReturnType<typeof setTimeout> | null;
+    let context: this | null = this;
+    let timestamp: number;
+    let result: void;
 
     const later = () => {
       let last = new Date().getTime() - timestamp;
       if (last < millisecsToWait) {
         timeout = setTimeout(later, millisecsToWait - last);
       } else {
         timeout = null;
-        result = func.apply(context, args);
+        result = func.apply(context);
```

Comment:
Just noticed this, why are args removed there?

Reply:
Because args was not getting any value here and it was showing error again and again that " [ ] can not be assigned to IArguments type".
I tried assigning other types also to args but then it was showing error that this type cannot be assigned to IArguments type variable and as args was not getting any value so I removed it.

Reply:
> Just noticed this, why are args removed there?

Because args was not getting any value here and it was showing error again and again that " [ ] can not be assigned to IArguments type".
I tried assigning other types also to args but then it was showing error that this type cannot be assigned to IArguments type variable and as args was not getting any value so I removed it.

Reply:
We should not remove the params here because we aren't able to find the correct type. 

Have you tried checking the value of args by manually running the server and trying to cover this function? OR maybe check where this `debounce` function is used in the codebase, I think that will be helpful.

Reply:
Yes, I checked the directive where ```debounce``` is used. It is used only at one place in codebase in ```top-navigation-bar.directive.ts``` on line Number: 333 and after that, I got to know that function doesn't take any arguments, therefore, removed it.